### PR TITLE
Add generic cancel task action

### LINKED ISSUE #8701: Add ability to cancel tasks in DB
Certain tasks in the appeals process are canceled/resolved/closed rather than completed. Two examples of this are:
- If a Veteran withdraws their hearing request, the Schedule Veteran task is closed
- If a Veteran waives their 90-day evidence window, the 90-day Evidence window is closed

Note: this is different than unassigning a task that was assigned to you, making the top level task back the team queue task, but LP wonders if there is any similarity. 

## Acceptance criteria
* [x] Update `closed_at` timestamp when task is cancelled or completed (PR #9312)
* [x] Add method that does not differentiate between a task being completed successfully and being abandoned in some other way using a new `Task.active?` method (PR #9169)
* [x] Update existing code to use the new `Task.active?` method where applicable (PR #9359)
* [ ] Add backend infrastructure to support cancelling of tasks (PR #9373)
* [ ] Update existing code to set the status to `cancelled` instead of `completed` where applicable (PR #9373)
* [x] Drop the `completed_at` column from the `tasks` table (PR #9181)

### PR DESCRIPTION
**Pull request by lowellrex**
Resolves #8701. Adds the "Cancel task" option to the list of actions available for `GenericTask`s in preparation for it being added to other tasks. Changes some existing places where we set the status of tasks to `completed` to `cancelled` instead.

full-flow

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -603,7 +603,7 @@ def other_close_event_date
     return if active_status?
     return if decision_issues.any?
 
-    root_task.completed_at
+    root_task.closed_at
```
Comment (lowellrex) [inline app/models/appeal.rb]: `completed_at` field [no longer exists](https://github.com/department-of-veterans-affairs/caseflow/pull/9181) in the `tasks` table. Don't know why this wasn't causing any test failures...

File: app/models/task.rb [hunk 2]
```diff
@@ -13,8 +13,8 @@ class Task < ApplicationRecord
   after_create :put_parent_on_hold
 
   before_update :set_timestamps
-  after_update :update_parent_status, if: :status_changed_to_completed_and_has_parent?
-  after_update :update_children_status, if: :status_changed_to_completed?
+  after_update :update_parent_status, if: :status_changed_to_closed_and_has_parent?
+  after_update :update_children_status, if: :status_changed_to_closed?
```
Comment (lowellrex) [inline app/models/task.rb]: We are treating the cancellation of tasks the same as the completion of tasks with regards to how that action bubbles up and down through parent and child tasks. [Related slack discussion](https://dsva.slack.com/archives/C6E41RE92/p1550001536061900) where we settled on this being the correct behaviour.

File: app/models/task.rb [hunk 3]
```diff
@@ -210,7 +210,7 @@ def reassign(reassign_params, current_user)
       t.save!
     end
 
-    update!(status: Constants.TASK_STATUSES.completed)
+    update!(status: Constants.TASK_STATUSES.cancelled)
```
Comment (lowellrex) [inline app/models/task.rb]: Reassigning tasks now cancels the old task.

File: app/models/tasks/schedule_hearing_task.rb [hunk 4]
```diff
@@ -97,9 +97,8 @@ def update_from_params(params, current_user)
 
         slot_new_hearing(hearing_day_id, hearing_type, hearing_time, hearing_location)
         HoldHearingTask.create_hold_hearing_task!(appeal, parent)
-      elsif params[:status] == "canceled"
+      elsif params[:status] == Constants.TASK_STATUSES.cancelled
```
Comment (lowellrex) [inline app/models/tasks/schedule_hearing_task.rb]: Now that we have the `cancelled` task status we can use it here.

File: app/models/task.rb [hunk 5]
```diff
@@ -394,12 +403,12 @@ def update_parent_status
...
-    saved_change_to_attribute?("status") && completed?
+  def status_changed_to_closed?
+    saved_change_to_attribute?("status") && !active?
   end
 
-  def status_changed_to_completed_and_has_parent?
-    status_changed_to_completed? && parent
+  def status_changed_to_closed_and_has_parent?
+    status_changed_to_closed? && parent
   end
```
Comment (tomas-nava) [inline app/models/task.rb]: These names (`status_changed_to_closed?` and `status_changed_to_closed_and_has_parent?`) feel misleading to me now; `task_closed?` and `task_closed_and_has_parent?` would be more accurate.

Comment (lowellrex) [reply app/models/task.rb]: Agreed! Maybe we could go with `task_just_closed?` to convey that the status just changed? `task_closed?` doesn't imply to me as strongly that we are checking to see if a status change just occurred.

**Pull request merged**